tests/test_config.py

```diff
@@ -1024,6 +1026,64 @@ def test_to_datafile_from_bytes(self):
 
         self.assertEqual(expected_datafile, actual_datafile)
 
+    def test_datafile_with_audience_segments(self):
+        """ Test to confirm that integration conversion works and has expected output """
+        opt_obj = optimizely.Optimizely(
+            json.dumps(self.config_dict_with_audience_segments)
+        )
+        project_config = opt_obj.config_manager.get_config()
+        self.assertIsInstance(project_config, ProjectConfig)
+
+        for integration in project_config.integration_key_map.values():
+            self.assertIsInstance(integration, entities.Integration)
+
+        integrations = self.config_dict_with_audience_segments['integrations']
+        self.assertGreater(len(integrations), 0)
+        self.assertEqual(len(project_config.integrations), len(integrations))
+
+        integration = integrations[0]
+        self.assertEqual(project_config.host_for_odp, integration['host'])
+        self.assertEqual(project_config.public_key_for_odp, integration['publicKey'])
+
+        self.assertEqual(sorted(project_config.all_segments), ['odp-segment-1', 'odp-segment-2', 'odp-segment-3'])
+
+    def test_datafile_with_no_audience_segments(self):
+        """ Test to confirm that datafile with empty integrations still works """
+        config_dict_with_audience_segments = copy.deepcopy(self.config_dict_with_audience_segments)
+        config_dict_with_audience_segments['integrations'] = []
+        opt_obj = optimizely.Optimizely(
+            json.dumps(config_dict_with_audience_segments)
+        )
+
+        project_config = opt_obj.config_manager.get_config()
+
+        self.assertIsInstance(project_config, ProjectConfig)
+        self.assertEqual(len(project_config.integrations), 0)
+
+    def test_datafile_with_audience_segments_missing_key(self):
+        """ Test to confirm that datafile with empty integrations still works """
+        config_dict_with_audience_segments = copy.deepcopy(self.config_dict_with_audience_segments)
+        del config_dict_with_audience_segments['integrations'][0]['host']
+        opt_obj = optimizely.Optimizely(
+            json.dumps(config_dict_with_audience_segments)
+        )
+
+        project_config = opt_obj.config_manager.get_config()
+
+        self.assertIsNone(project_config)
+
+    def test_datafile_with_invalid_audience_segments(self):
```

Comment:
Can we replace this test with "key" only integration (missing other fields), and it's accepted ok?
The only field in "integrations" is "key", while all others optional.